relay-server/src/services/buffer/mod.rs

```diff
@@ -169,6 +173,11 @@ impl EnvelopeBufferService {
             tokio::time::sleep(self.sleep).await;
         }
 
+        // In case the project cache is not ready, we don't want to attempt popping.
+        if !self.project_cache_ready.load(Ordering::Relaxed) {
+            return future::pending().await;
```

Comment:
can we do a sleep for `DEFAULT_SLEEP` here? If Relay is in a mode where it does not receive any messages, this might block forever.

Reply:
Good point!

Reply:
The thing though is that, if the buffer is not ready to accept data and no messages are coming in, what should relay do?